### PR TITLE
[RAPPS] Add localized display of installation date and time

### PR DESCRIPTION
**Pull request by BUGinator86**
We want the datetime formated in local datetime independent from the source registry-key

## Purpose

_Show install time of software in RAPPS as localized datetime instead of YYYYDDMM formatted string_

JIRA issue: [CORE-17422](https://jira.reactos.org/browse/CORE-17422)

### CODE REVIEW

**Comment by binarymaster** [comment]
Before merging you need to set up your GitHub account correctly, so maintainers can use Squash and Merge button without getting the author field modified back to nickname by GitHub, it's done in two steps:

1. Set your full name in [GitHub profile settings](https://github.com/settings/profile)
1. The email address in your commits does not seem to be attached with your GitHub account, please attach it, it's in [GitHub email settings](https://github.com/settings/emails)

Regards,

**Comment by BUGinator86** [comment]
> Before merging you need to set up your GitHub account correctly, so maintainers can use Squash and Merge button without getting the author field modified back to nickname by GitHub, it's done in two steps:
> 
> 1. Set your full name in [GitHub profile settings](https://github.com/settings/profile)
> 2. The email address in your commits does not seem to be associated with your GitHub account, please associate it in [GitHub email settings](https://github.com/settings/emails)
> 
> Regards,

Done :)

**Comment by binarymaster** [comment]
Looks correct now, thanks!

**Review by learn-more — APPROVED** [review]

File: base/applications/rapps/installed.cpp
Comment (HBelusca) [inline base/applications/rapps/installed.cpp:55]:
```suggestion
            // Check if we have 8 characters to parse the datetime.
            // Maybe other formats exist as well?
```

**Review by HBelusca — APPROVED** [review]

**Review by binarymaster — COMMENTED** [review]
I just tested it, on both ReactOS and Windows, doesn't seem to display localized date for me:

image

image

File: base/applications/rapps/installed.cpp
Comment (binarymaster) [inline base/applications/rapps/installed.cpp:56]: In my both cases `szInstallDate.GetLength()` returns 9, the last character is UNICODE_NULL.

Comment (binarymaster) [inline base/applications/rapps/installed.cpp:56]: This small addition fixes it for me:
```suggestion
            szInstallDate = szInstallDate.Trim();
            if (szInstallDate.GetLength() == 8)
```

**Review by binarymaster — APPROVED** [review]
LGTM now:

VirtualBox_ReactOS_22_05_2022_20_26_37

**Pull request merged**
